hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/TestSqlSource.java

```diff
@@ -135,6 +135,23 @@ public void testSqlSourceRowFormat() throws IOException {
     assertEquals(10000, fetch1AsRows.getBatch().get().count());
   }
 
+  /**
+   * Runs the test scenario of reading data from the source in row format.
+   * Source has no records.
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testSqlSourceCheckpoint() throws IOException {
```

Comment:
It would be good to have a test for SqlSource in TestHoodieDeltaStreamer as well. You can take it up as a follow-up taks if you like. Typically, for every source we want to have a unit test and a deltastreamer test.